File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/C_Factors_of_Factorial.cpp
#include <bits/stdc++.h>
#define MOD 1'000'000'007
using namespace std;
typedef long long ll;

ll N;

void factorize(ll n, map<ll, ll> &C) {
  for (int i = 2; i * i <= n; ++i) {
    if (n % i != 0) continue;
    while (n % i == 0) {
      n /= i;
      ++C[i];
    }
  }
  if (n > 1) {
    ++C[n];
  }
}

void solve() {
  map<ll, ll> C;
  for (int i = 1; i <= N; ++i) {
    factorize(i, C);
  }
  ll res = 1;
  for (auto p : C) {
    res = res * (p.second + 1) % MOD;
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/C_Factors_of_Factorial.py
from collections import Counter
MOD = 10**9 + 7
N = int(input())


def factorize(n, C):
    for i in range(2, int(n**0.5) + 1):
        if n % i != 0:
            continue
        while n % i == 0:
            n //= i
            C[i] += 1
    if n > 1:
        C[n] += 1


C = Counter()
for i in range(2, N + 1):
    factorize(i, C)
res = 1
for v in C.values():
    res = res * (v + 1) % MOD
print(res)
